packages/blockchain-wallet-v4-frontend/src/modals/SimpleBuy/CheckoutConfirm/template.success.tsx

```diff
@@ -70,15 +70,15 @@ const Success: React.FC<InjectedFormProps<{}, Props> & Props> = props => {
   const baseCurrency = getBaseCurrency(props.order, props.supportedCoins)
   const counterAmount = getCounterAmount(props.order)
   const counterCurrency = getCounterCurrency(props.order, props.supportedCoins)
+  const requiresTerms =
+    props.order.paymentType === 'PAYMENT_CARD' ||
+    props.order.paymentType === 'USER_CARD'
 
-  if (
-    !(
-      props.order.paymentType === 'PAYMENT_CARD' ||
-      props.order.paymentType === 'USER_CARD'
-    )
-  ) {
-    setAcceptTerms(true)
-  }
+  useEffect(() => {
+    if (!requiresTerms) {
+      return setAcceptTerms(true)
+    }
+  }, [])
```

Comment:
here we can pass `acceptTerms` and then it will run useEffect only in case it has been changed

Reply:
Why would we need to do that?

Reply:
that way we said to react skip applying it unless it has been changed

Reply:
but not mandatory at all, just a note